__tests__/app.test.js

```diff
@@ -0,0 +1,387 @@
+          expect(typeof comment.article_id).toBe("number");
+          expect(typeof comment.votes).toBe("number");
+          expect(typeof comment.body).toBe("string");
+          expect(typeof comment.author).toBe("string");
+          expect(typeof comment.created_at).toBe("string");
+        });
+    });
+    test("Returns 400 Bad Request if given an article_id that is invalid", () => {
+      const postData = { body: "example body", username: "butter_bridge" }
+      return request(app)
+      .get("/api/articles/soap/comments")
+      .send(postData)
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("Bad Request");
+        });
+    });
+    test("Returns 404 not found if given a valid article_id that does not exist", () => {
+      const postData = { body: "example body", username: "butter_bridge" }
+      return request(app)
+        .post("/api/articles/12345/comments")
+        .send(postData)
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("Not Found");
+        });
+    });
+    test("Returns a 401 when given an invalid username", ()=>{
+      const postData = { body: "example body", username: "not_a_user" }
+      return request(app)
+      .post("/api/articles/2/comments")
+        .send(postData)
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("Bad Request");
+        });
+    });
+  });
+});
+
+describe("/api/comments/:comment_id", () => {
+  describe("GET", () => {
+    test("Selects a comment by given comment_id", () => {
+      return request(app)
+        .get("/api/comments/1")
+        .expect(200)
+        .then((response) => {
+          const comment = response.body.comment
+          expect(comment.comment_id).toBe(1)
+          expect(comment.body).toBe("Oh, I've got compassion running out of my nose, pal! I'm the Sultan of Sentiment!")
+          expect(comment.article_id).toBe(9)
+          expect(comment.author).toBe('butter_bridge')
+          expect(comment.votes).toBe(16)
+          expect(comment.created_at).toBe('2020-04-06T12:17:00.000Z')
+        });
+    });
+    test("Returns 404 when given valid comment_id that does not exist", ()=>{
+      return request(app)
+      .get("/api/comments/10000")
+      .expect(404)
+      .then((response)=>{
+        expect(response.body.msg).toBe('Not Found')
+      })
+
+    })
+    test("Returns 400 when given invalid comment_id", ()=>{
+      return request(app)
+      .get("/api/comments/gorilla")
+      .expect(400)
+      .then((response)=>{
+        expect(response.body.msg).toBe('Bad Request')
+      })
+
+    })
+
+  });
+  describe("DELETE", () => {
+    test("Deletes comment at given comment_id and responds with 204 and empty response", () => {
+      return request(app)
+        .delete("/api/comments/1")
+        .expect(204)
+        .then((response) => {
+          expect(response.body).toEqual({});
```

Comment:
We don't actually need to check this as a 204 infers there will be nothing on the response body